jdaviz/configs/mosviz/tests/test_parsers.py

```diff
@@ -27,11 +27,14 @@ def test_nirspec_parser(mosviz_helper, tmp_path, instrument_arg):
     if instrument_arg:
         mosviz_helper.load_data(directory=data_dir, instrument=instrument_arg)
     else:
-        # When no instrument is provided, mosviz.load_data should fallback to the nirspec loader.
-        # Naturally, the nirspec dataset should then work without any instrument keyword
-        with pytest.warns(UserWarning, match="Ambiguous MOS Instrument"):
+        # When no instrument is provided, Mosviz now raises an error
+        match = ('Ambiguous MOS Instrument: Only JWST NIRSpec, NIRCam, and NIRISS folder parsing'
+                 ' are currently supported')
+        with pytest.raises(ValueError, match=match):
             mosviz_helper.load_data(directory=data_dir)
 
+        return
```

Comment:
It feels unnatural to `return` in a test function. Maybe this is a sign that parametrize is not suitable for this test case anymore. You don't need remote data to prove that the function will throw an exception, right? Theoretically, you could give a dummy file name/object and it would still throw the same exception?

Anyways, if we want to chuck this in the technical debt bin, we can refactor in the future and leave this as-is.

Reply:
Into the bin!